src/libraries/System.Diagnostics.DiagnosticSource/src/System/Diagnostics/Metrics/Instrument.netcore.cs

```diff
@@ -0,0 +1,75 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Collections.Generic;
+using System.Runtime.InteropServices;
+using System.Runtime.CompilerServices;
+
+namespace System.Diagnostics.Metrics
+{
+    /// <summary>
+    /// Instrument{T} is the base class from which all non-observable instruments will inherit from.
+    /// </summary>
+    /// <remarks>
+    /// This class supports only the following generic parameter types: <see cref="byte" />, <see cref="short" />, <see cref="int" />, <see cref="long" />, <see cref="float" />, <see cref="double" />, and <see cref="decimal" />
+    /// </remarks>
+    public abstract partial class Instrument<T> : Instrument where T : struct
+    {
+
+        [StructLayout(LayoutKind.Sequential)]
+        private struct TagsBag
```

Comment:
This nested type will get an implicit `T` and so it will be duplicated for each `Instrument<T>` unnecessarily. Move it out of the generic type to avoid this problem.

Reply:
good catch. I'll move it out. Thanks.